example-get-started-experiments/code/sagemaker/deploy_model.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import re
+import sys
+
+from sagemaker.deserializers import JSONDeserializer
+from sagemaker.pytorch import PyTorchModel
+from sagemaker.serverless import ServerlessInferenceConfig
+
+
+memory_size = { 
+    "dev": 1024,
+    "staging": 1024,
+    "prod": 2048,
+    "default": 1024,
+}
+max_concurrency = { 
+    "dev": 5,
+    "staging": 5,
+    "prod": 10,
+    "default": 5,
+}
+
+
+def deploy(
+    name: str,
+    stage: str,
+    version: str,
+    model_data: str,
+    role: str,
+):
+    sagemaker_logger = logging.getLogger("sagemaker")
+    sagemaker_logger.setLevel(logging.DEBUG)
+    sagemaker_logger.addHandler(logging.StreamHandler(sys.stdout))
+
+    composed_name =  re.sub(
+        r"[^a-zA-Z0-9\-]", "-", f"{name}-{version}-{stage}")
```

Comment:
Do you know what possible characters can appear in `f"{name}-{version}-{stage}"` that would need to be replaced here? I guess it's related to https://github.com/iterative/dvc/issues/9821?

Reply:
> I guess it's related to [iterative/dvc#9821](https://github.com/iterative/dvc/issues/9821)?

This regex is only relevant for Sagemaker and it is enforced by the API

Reply:
I think the rationale for the naming restrictions in gto were to make it more likely they would be compatible with systems like sagemaker. Not sure it really helps though. We still hit these restrictions and users don't know why they can't use these characters.